pkg/splunk/controller/statefulset.go

```diff
@@ -282,3 +282,37 @@ func GetStatefulSetByName(c splcommon.ControllerClient, namespacedName types.Nam
 
 	return &statefulset, nil
 }
+
+// VerfiyMCStatefulSetOwnerRef deletes the automated MC sts. This is when customer migrates from automated MC to MC CRD
+// Check if MC CR is not the owner of the MC statefulset then delete that Statefulset
+func VerfiyMCStatefulSetOwnerRef(client splcommon.ControllerClient, cr splcommon.MetaObject, namespacedName types.NamespacedName) error {
```

Comment:
Nit: Can we rename it to something that reflects we are deleting the old-style automatic MC pod
ex. deleteReferencesToAutomaticMcIfExists() ? or something like that.

Reply:
Nit: Typo